src/main/scala/codacy/scalastyle/ScalaStyle.scala

```diff
@@ -31,19 +34,16 @@ object ScalaStyle extends Tool {
         paths.map(_.toString).toList
       }
 
-      val configurationOption = List(
-        "--config",
-        //priorities: codacy-patterns then a native config in the project-root then the default config
-        getConfigFile(fullConfig).orElse(nativeConfigFile).getOrElse(defaultConfigFile).getAbsolutePath
-      )
+      val scalastyleConfiguration = {
+        val path = getConfigFile(fullConfig).orElse(nativeConfigFile).getOrElse(defaultConfigFile).getAbsolutePath
+        ScalastyleConfiguration.readFromXml(path)
```

Comment:
Does this can blow or something? I guess it is fine either way, since we have the big Try wrapping this?

Reply:
> Does this can blow or something?

Yes it can but it's supposed to, if the config file is wrong.

> I guess it is fine either way, since we have the big Try wrapping this?

Exactly 👌